Types/PDF/src/pdf.cpp

```diff
@@ -0,0 +1,621 @@
+    {
+        uint8_t buffer;
+        bool match = true;
+        for (uint64 i = 0; i < size_type; ++i) {
+            if (!data.Copy(offset + i, buffer) || buffer != PDF_ARRAY[i]) {
+                match = false;
+                break;
+            }
+        }
+        return match;
+    }
+
+    uint64 GetTypeValue(GView::Utils::DataCache& data, uint64& offset, const uint64& dataSize)
+    {
+        std::string lengthValStr;
+        uint8_t buffer;
+        uint64 value = 0;
+        while (offset < dataSize && data.Copy(offset, buffer) && buffer >= '0' && buffer <= '9') {
+            lengthValStr.push_back(buffer);
+            offset++;
+        }
+
+        if (!lengthValStr.empty()) {
+            value = std::stoull(lengthValStr);
+        }
+        return value;
+    }
+
+    uint8 GetWValue(GView::Utils::DataCache& data, uint64& offset)
+    {
+        std::string lengthValStr;
+        uint8_t buffer;
+        uint8 value = 0;
+        if (data.Copy(offset, buffer)) {
+            lengthValStr.push_back(buffer);
+            offset++;
+        }
+
+        if (!lengthValStr.empty()) {
+            value = std::stoull(lengthValStr);
+        }
+        return value;
+    }
+
+    void GetFilters(GView::Utils::DataCache& data, uint64& offset, const uint64& dataSize, std::vector<std::string> &filters)
+    {
+        uint8_t buffer;
+        std::string filterValue;
+        filterValue += "/";
+        if (data.Copy(offset, buffer) && buffer == PDF::WSC::SPACE) { // /Filter /
+            offset++;
+        }
+        offset++; // skip "/"
+        while (offset < dataSize) {
+            if (!data.Copy(offset, buffer)) {
+                break;
+            }
+            if (buffer == PDF::DC::SOLIUDS || buffer == PDF::DC::GREATER_THAN || buffer == PDF::WSC::LINE_FEED) {
+                break;
+            } else {
+                filterValue += static_cast<char>(buffer);
+                offset++;
+            }
+        }
+
+        if (filterValue.length() > 1) {
+            filters.push_back(filterValue);
+        }
+    }
+
+    void GetDecompressDataValue(Buffer& decompressedData, uint64& offset, const uint8& value, uint64& obj)
+    {
+        for (uint8_t i = 0; i < value; ++i) {
+            obj = (obj << 8) | decompressedData[offset + i];
+        }
+        offset += value;
+    }
+
+    void GetObjectsOffsets(const uint64& numEntries, uint64& offset, GView::Utils::DataCache& data, std::vector<uint64_t> &objectOffsets)
+    {
+        // Read each 20-byte entry
+        for (uint64 i = 0; i < numEntries; ++i) {
+            char entry[21];
+            memset(entry, 0, sizeof(entry));
+
+            for (size_t j = 0; j < 20; ++j) {
+                if (!data.Copy(offset + j, entry[j])) {
+                    break;
+                }
+            }
+
+            if (entry[17] != 'f') { // skip the free entries
```

Comment:
what is `f`? are there some constants? :D 

also 20 might be a constant as well